TACC/Core-Portal
# task/TV3-82: v3 Push Keys

## Overview

Modify related push keys components and methods for use with Tapis v3. Notably, https://github.com/TACC/Core-Portal/pull/740 removes a lot of now-defunct code surrounding app and system cloning, and puts focus on hostname vs system id when looking for compatible credentials.

## Related

* [TV3-82](https://jira.tacc.utexas.edu/browse/TV3-82)
* https://github.com/TACC/Core-Portal/pull/740

## Changes

- add tapis v3 push keys methods
- implement listType when getting apps
- remove non key service systems from onboarding
- add more sys onboarding logs


## Testing

Make sure you're using the latest credentials from stashe to get the `portals` tenant setup.

0. If you haven't already, add an app category and entry for the app `hello-world` in https://cep.test/core/admin
1. If you already have valid credentials for the `frontera` exec system on the `portals` develop tenant, run this with the `wma_prtl` user in this `portals` develop tenant to remove your credentials: `wma_prtl.systems.removeUserCredential(systemId=frontera, userName=your_username)`
2. Go to https://cep.test/workbench/applications/hello-world
3. Click on the button that asks you to push keys
4. Enter your credentials, and confirm it succeeds and does not prompt you to push keys again after reloading the app

## UI
![Screen Shot 2022-12-13 at 10 03 21 AM](https://user-images.githubusercontent.com/20326896/207395224-8b044d96-0d0e-4fd2-b38c-fa4004c602a6.png)

## Notes
Currently the `getApps` command is not returning any publicly shared apps, so the app tray tabs will appear empty.
TODO:
- [x] Reenable empty tab filtering code once Tapis `getApps` is fixed in `server/portal/apps/workspace/api/views.py`:`AppsTrayView`:`get`
